### DISCUSSION TITLE
Script - access to TEMPTARGETSET based on INDEX

### DISCUSSION THREAD

**Discussion by gck303** [arendst/Tasmota#11993] [Q&A] [answered]
I want to be able to access the thermostat values inside a script. I want to show the measured/target values on a display. 

How do I do this? I am at a loss. 

`>D
temp1=0
TEMPTARGETSET=0
displaydimmer 15
;=>power1 1

>E
temp1=TEMPTARGETSET
print temp1: %TEMPTARGETSET% %temp1%

>T
temp1=TEMPTARGETSET

>S
dt[y0]Target:%0temp1% %TEMPTARGETSET% `

**Comment by sfromis** [comment, marked as answer]
I know really nothing about scripting, but I'd avoid having a variable with the same name as the trigger (`TEMPTARGETSET`) you're trying to get working. If the name resolution looks at variables before triggers, that could cause the problem you see. Thus, I'd suggest removing `TEMPTARGETSET=0` from the `>D` section. Don't know if it will work, but why not try it.....

**Reply by gck303** [reply]
Thank you. That worked and it will now allow me to get both the values.

`>D
temp1=0
set1=0
;TEMPTARGETSET=0
displaydimmer 15
;=>power1 1

>E
temp1=TEMPMEASUREDSET
set1=TEMPTARGETSET
print temp1: %temp1% 
print set1: %set1%

>T
;temp1=TEMPTARGETSET

>S
dt[y0]Target:%0temp1% %0set1%

`

**Reply by gck303** [reply]
The answer given by @sfromis only partially addresses my issue. It allows me to get the most recently updated value, but it will not allow me to get the value based on the index. How can this be done?

I need to get the temp based on the thermostat number. I tried using square brackets, but it  did not work. 

`>D
temp1=0
set1=0
dp1=0
temp2=0
set2=0
dp2=0
displaydimmer 15
;=>power1 1
>E
temp1=TEMPMEASUREDSET[1]
set1=TEMPTARGETSET[1]
temp2=TEMPMEASUREDSET[2]
set2=TEMPTARGETSET[2]
print temp1: %temp1% temp2: %temp2%
print set1: %set1% set2: %set2%
>S
dt[y0p20]T:%0temp1% %0temp2%
dt[y1p20]S:%0set1% %0set2%
`
